python/tests/integration/connectors/memory/test_pinecone.py

```diff
@@ -23,6 +24,7 @@
 async def retry(func, retries=1):
     for i in range(retries):
         try:
+            time.sleep(3)
```

Comment:
this makes the whole test suite more likely to pass by giving some time between pinecone operations. Although it is not perfect.